[FINGERPRINT] common: Use new power and IRQ IOCTLs for FPC device driver

kholk opened the pull request:
The kernel driver now features a miscdevice and IOCTLs.
Add definitions for all of the IOCTLs and convert the
device power management to use them.

This also unifies the power on/off functions between the
implementations.

------------------------------------------------------------------------------
This patchset is not yet complete.
Need to finish the implementation with the IRQ IOCTL, plus a
bugfix for the Yoshino platform crash which is currently under
testing and not yet published.

This PR is there as a partial showdown of what will happen in
the near future, to make other users able to eventually help
testing the new IOCTL implementation in the kernel driver.

This code will only work with this kernel related PR:
https://github.com/sonyxperiadev/kernel/pull/1518

Please note that this commit may change, as it may be better
to actually pull an IOCTL definition header from the kernel itself.

kholk committed:
common: Use new power up/down IOCTLs for FPC device driver

The kernel driver now features a miscdevice and IOCTLs.
Add definitions for all of the IOCTLs and convert the
device power management to use them.

This also unifies the power on/off functions between the
implementations.

kholk commented:
Forcepushed: Corrected fpc_{get,set}_power IOCTL error condition.

alviteri commented on common.h:
```diff
@@ -3,7 +3,22 @@
+enum {
+    FPC_PWROFF = 0,
+    FPC_PWRON = 1,
+}
```
hardware/sony/fingerprint/common.h:17:2: error: expected ';' after enum
}
 ^
1 error generated.

kholk committed:
common: Add support for IOCRIRQPOLL IOCTL

This IOCTL polls for FPC interrupt.

kholk committed:
fpc_imp: L-T-Y: Allocate and use global buffer for normal command

As also stated by the TODO comment, allocating and freeing the
buffer again and again is useless and only makes us waste
precious CPU cycles and bandwidth because of the continuously
called memsets.

Because that buffer is always of a known 0x40 length, implement
a global ION buffer and use it for the normal command.

kholk committed:
fpc_imp: L-T-Y: Use IOCTL interrupt polling

After implementing the common function to manage the new
IOCRIRQPOLL IOCTL, switch the supported implementations
to the new handling.

kholk commented:
Forcepushed: Fixed one build error, implemented support for IRQ polling via IOCTLs, switched Loire, Tone, Yoshino to the new handling.

kholk commented:
Forcepushed: Fix Loire/Tone implementation (moved ION allocation), fixed typo
